src/init.cpp

```diff
@@ -1091,11 +1091,11 @@ static bool LockDataDirectory(bool probeOnly)
     return true;
 }
 
-bool AppInitSanityChecks()
+bool AppInitSanityChecks(NodeContext& node)
 {
     // ********************************************************* Step 4: sanity checks
 
-    init::SetGlobals();
+    node.kernel = std::make_unique<kernel::Context>();
```

Comment:
Mmm, this whole function is a mess and doesn't really make much sense anymore.

I really don't like the creation of the `kernel::Context` happening in here. That's not at all intuitive. I would much prefer to see the caller create and set it. Mind addressing that here?

A few potential follow-ups for a later PR:
- I'm hoping that the SanityChecks move into the `kernel::Context` creation as a next step.
- I think it may make sense for `NodeContext` to take a `kernel::Context` in its constructor so that it always exists in the NodeContext. That might be much easier said than done though, I haven't looked deeply into it.
- Those changes would leave only `LockDataDirectory` here, meaning that `AppInitSanityChecks` would just go away.

Reply:
> I would much prefer to see the caller create and set it. Mind addressing that here?

Let me know what you think about: 8d63a934cf4053851ba79d4a44326cd079bc8288

Reply:
> A few potential follow-ups for a later PR:
> ...

I will open an issue after this one's merged so we don't lose track of these!

Reply:
Ah yes, this is much more straightforward imo. +1 for squashing that in.